packages/core/src/agent/index.ts

```diff
@@ -3265,19 +3268,19 @@ Message ${msg.threadId && msg.threadId !== threadObject.id ? 'from previous conv
       throw error;
     }
 
-    return fullOutput as ReturnType<
-      FORMAT extends 'mastra' ? MastraModelOutput['getFullOutput'] : AISDKV5OutputStream['getFullOutput']
-    >;
+    return fullOutput as unknown as FORMAT extends 'aisdk'
+      ? Awaited<ReturnType<AISDKV5OutputStream<OUTPUT>['getFullOutput']>>
+      : Awaited<ReturnType<MastraModelOutput['getFullOutput']>>;
```

Comment:
Should we also be passing OUTPUT to `MastraModelOutput`? since that has `object` as well.

Reply:
object in base output is the same as aisdk output. The aisdk output getter just returns the same one.